### ISSUE TITLE
selinux: properly implement selinux access control for unit install operations

### ISSUE THREAD

**Issue by poettering** [systemd/systemd#1050]
We need to properly fix selinux access control for unit install operations, so that the selinux access check is done from inside install.c right after figuring out what to do and having searched the unit files, and right before actually doing them.

This is a follow-up issue from:

https://github.com/systemd/systemd/pull/1044#issuecomment-135036652

Just to keep track of the problem, and so that we remember to fix this properly.

**Comment by ghost** [comment]
And this is the right way to do it? To just drop stuff right before the release?

**Comment by poettering** [comment]
@doverride yes, the version in the last release broke things for both non-SELinux and SELinux systems, so it couldn't stay that way. And a proper fix is not that easy. Given that SeLinux is just second level security it is OK to remove this for now, as long as we fix this properly eventually. Hence this bug. Of course, distros which care about SELinux should probably not base their stable version on this one.

**Comment by xnox** [comment]
This is only a regression iff the selinux support is compiled, right? And w.r.t. to which systemd release is this a regression (as in what was the last known good one)?
